src/sentry/static/sentry/app/components/confirm.tsx

```diff
@@ -135,149 +124,122 @@ class Confirm extends React.PureComponent<Props, State> {
+
+    onConfirming?.();
+    this.setState({disableConfirmButton: disableConfirmButton || false});
+
+    // always reset `confirming` when modal visibility changes
+    this.confirming = false;
+
+    this.activateModal();
   };
 
-  render() {
-    const {
-      disabled,
-      message,
-      renderMessage,
-      priority,
-      confirmText,
-      cancelText,
-      children,
-      header,
-    } = this.props;
+  activateModal = () =>
+    openModal(({Header, Body, Footer, closeModal}) => {
+      const {
+        priority,
+        renderMessage,
+        message,
+        confirmText,
+        cancelText,
+        header,
+        onConfirm,
+        onCancel,
+        disableConfirmButton,
+      } = this.props;
+
+      const handleClose = () => {
+        onCancel?.();
+        this.setState({disableConfirmButton: disableConfirmButton || false});
+
+        // always reset `confirming` when modal visibility changes
+        this.confirming = false;
+        closeModal();
+      };
 
-    let confirmMessage: React.ReactNode;
+      const handleConfirm = () => {
+        // `confirming` is used to make sure `onConfirm` is only called once
+        if (!this.confirming) {
+          onConfirm();
+        }
 
-    if (typeof renderMessage === 'function') {
-      confirmMessage = renderMessage({
-        confirm: this.handleConfirm,
-        close: this.handleToggle,
-      });
-    } else {
-      confirmMessage = React.isValidElement(message) ? (
-        message
-      ) : (
-        <p>
-          <strong>{message}</strong>
-        </p>
-      );
-    }
+        this.setState({disableConfirmButton: true});
+        this.confirming = true;
+        closeModal();
+      };
 
-    return (
-      <React.Fragment>
-        {typeof children === 'function'
-          ? children({
-              close: this.closeModal,
-              open: this.openModal,
-            })
-          : React.isValidElement(children) &&
-            React.cloneElement(children, {
-              disabled,
-              onClick: this.handleToggle,
-            })}
-        <Modal show={this.state.isModalOpen} animation={false} onHide={this.handleToggle}>
-          {header && <Modal.Header>{header}</Modal.Header>}
-          <Modal.Body>{confirmMessage}</Modal.Body>
-          <Modal.Footer>
-            <Button style={{marginRight: 10}} onClick={this.handleToggle}>
+      const confirmMessage =
+        typeof renderMessage === 'function' ? (
+          renderMessage({
+            confirm: handleConfirm,
+            close: handleClose,
+          })
+        ) : React.isValidElement(message) ? (
+          message
+        ) : (
+          <p>
+            <strong>{message}</strong>
+          </p>
+        );
+
+      return (
+        <React.Fragment>
+          {header && <Header>{header}</Header>}
+          <Body>{confirmMessage}</Body>
+          <Footer>
+            <Button style={{marginRight: 10}} onClick={handleClose}>
```

Comment:
can we style this button using emotion and our space utility?

Reply:
I will take care of this once I have all the tests passing (🙏 I do now)